src/com/sap/piper/JenkinsController.groovy

```diff
@@ -0,0 +1,115 @@
+package com.sap.piper
+
+class JenkinsController implements Serializable {
+    def script
+    String jenkinsUrl
+    def timeout
+
+    JenkinsController(script, String jenkinsUrl = "http://localhost:8080", timeout = 3600) {
+        this.script = script
+        this.jenkinsUrl = jenkinsUrl
+        this.timeout = timeout
+    }
+
+    def waitForJenkinsStarted() {
+        def timeout = 120
+        def timePerLoop = 5
+
+        for (int i = 0; i < timeout; i += timePerLoop) {
+            script.sleep timePerLoop
+            try {
+                if (retrieveJenkinsStatus() == 'NORMAL') {
+                    return true
+                }
+            } catch (Exception e) {
+                script.echo "Could not retrieve status for Jenkins. Message: ${e.getMessage()}. Retrying..."
+                continue
+            }
+            return false
+        }
+        script.error("Timeout: Jenkins did not start within the expected time frame.")
+    }
+
+    private retrieveJenkinsStatus() {
```

Comment:
You could handle the exception directly in this method and by that always return a valid value.